File: src/main/java/org/jukeboxmc/player/skin/SkinAnimation.java
package org.jukeboxmc.player.skin;

import lombok.ToString;

/**
 * @author LucGamesYT
 * @version 1.0
 */
@ToString
public class SkinAnimation {

    private final Image image;
    private final int type;
    private final float frames;
    private final int expression;

    public SkinAnimation( Image image, int type, float frames, int expression ) {
        this.image = image;
        this.type = type;
        this.frames = frames;
        this.expression = expression;
    }

    public Image getImage() {
        return this.image;
    }

    public int getType() {
        return this.type;
    }

    public float getFrames() {
        return this.frames;
    }

    public int getExpression() {
        return this.expression;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/player/skin/SkinAnimation.kt
package org.jukeboxmc.api.player.skin

/**
 * Represents a skin animation e.g. a twinkle in the eye
 */
class SkinAnimation(
    val skinImage: SkinImage,
    val type: Int,
    val frames: Float,
    val expression: Int
) {
    override fun toString(): String {
        return "SkinAnimation(skinImage=$skinImage, type=$type, frames=$frames, expression=$expression)"
    }
}